testbed/tests/window/test_dialogs.py

```diff
@@ -9,6 +11,28 @@
 TESTS_DIR = Path(__file__).parent.parent
 
 
+@pytest.fixture
+async def wait_for_dialog_to_close(main_window):
+    """Wait for any asyncio task that is responsible for closing the dialog.
+
+    An automatic fixture verifies tasks are not unintentionally left stranded after
+    the test; so, wait specifically for the task that closes the dialog before leaving.
+    """
+    yield
+    tasks = [
+        t for t in main_window.app._running_tasks if t.get_name() == "close-dialog"
+    ]
+    task = tasks[0] if tasks else None
+    deadline = time() + 1.5
+    while task and not task.done():
+        print("Waiting for dialog to close...")
+        await asyncio.sleep(0.1)
+        if time() > deadline:
+            print("Gave up waiting for dialog to close...")
+            break
+
+
+@pytest.mark.usefixtures("wait_for_dialog_to_close")
```

Comment:
Is there a functional difference between this, and using `wait_for_dialog_to_close` as an argument?

Reply:
No; they are functionally identical.

Reply:
In which case - personal preference would be to keep the syntax simpler. There's a bunch of other places in the code where fixtures are defined as arguments, but not actually used (cf the usage of `app` in a bunch of tests). Having more than one syntax for "use fixture X" grates against me a bit. 

However, if you feel strongly to the contrary, I won't fight too hard.

Reply:
I don't feel strongly; I might argue this is the simpler (or at least clearer) approach. The idiom of invoking a side-effect through including an argument in a function signature isn't really intuitive to me...however, I'd fully expect a decorator to have side-effects that might be interesting....(even there, though, I would have preferred a better name than "fixture"...but alas...).

Reply:
I agree that the "side effect" API style isn't great - that's one of the biggest criticisms I have of pytest as a framework. A version of pytest that used explicit declaration of fixtures (and allowed you do distinguish between the name of the fixture and the name that is passed to the test) would win me over almost immediately.

However, as it stands, pytest *does* do everything by implicit argument handling, so consistency with local (albeit annoying) condition wins out, in my book.